src/test/coins_tests.cpp

```diff
@@ -878,4 +895,209 @@ BOOST_AUTO_TEST_CASE(ccoins_write)
+void TestFlushBehavior(
+    CCoinsViewCacheTest* view,
+    CCoinsViewDB& base,
+    std::vector<CCoinsViewCacheTest*>& all_caches,
+    bool do_erasing_flush)
+{
+    CAmount value;
+    char flags;
+    uint256 txid;
+    COutPoint outp;
+    Coin coin;
+    size_t cache_usage;
+
+    auto flush_all = [&all_caches](bool erase) {
+        // Flush in reverse order to ensure that flushes happen from children up.
+        for (auto i = all_caches.rbegin(); i != all_caches.rend(); ++i) {
+            auto cache = *i;
+            // hashBlock must be filled before flushing to disk; value is
+            // unimportant here. This is normally done during connect/disconnect block.
+            cache->SetBestBlock(InsecureRand256());
+            erase ? cache->Flush() : cache->Sync();
+        }
+    };
+
+    txid = InsecureRand256();
+    outp = COutPoint(txid, 0);
+    coin = MakeCoin();
+    // Ensure the coins views haven't seen this coin before.
+    BOOST_CHECK(!base.HaveCoin(outp));
+    BOOST_CHECK(!view->HaveCoin(outp));
+
+    // --- 1. Adding a random coin to the child cache
+    //
+    Coin tmpCoin{coin};
+    view->AddCoin(outp, std::move(tmpCoin), false);
+
+    cache_usage = view->DynamicMemoryUsage();
+    // `base` shouldn't have coin (no flush yet) but `view` should have cached it.
+    BOOST_CHECK(!base.HaveCoin(outp));
+    BOOST_CHECK(view->HaveCoin(outp));
+
+    GetCoinsMapEntry(view->map(), value, flags, outp);
+    BOOST_CHECK_EQUAL(value, coin.out.nValue);
+    BOOST_CHECK_EQUAL(flags, DIRTY|FRESH);
+
+    // --- 2. Flushing all caches (without erasing)
+    //
+    flush_all(/*erase*/ false);
+
+    // CoinsMap usage should be unchanged since we didn't erase anything.
+    BOOST_CHECK_EQUAL(cache_usage, view->DynamicMemoryUsage());
+
+    // --- 3. Ensuring the entry still exists in the cache and has been written to parent
+    //
+    GetCoinsMapEntry(view->map(), value, flags, outp);
+    BOOST_CHECK_EQUAL(value, coin.out.nValue);
+    BOOST_CHECK_EQUAL(flags, 0);  // Flags should have been wiped.
+
+    // Both views should now have the coin.
+    BOOST_CHECK(base.HaveCoin(outp));
+    BOOST_CHECK(view->HaveCoin(outp));
+
+    if (do_erasing_flush) {
+        // --- 4. Flushing the caches again (with erasing)
+        //
+        flush_all(/*erase*/ true);
+
+        // Memory usage should have gone down.
+        BOOST_CHECK(view->DynamicMemoryUsage() < cache_usage);
+
+        // --- 5. Ensuring the entry is no longer in the cache
+        //
+        GetCoinsMapEntry(view->map(), value, flags, outp);
+        BOOST_CHECK_EQUAL(value, ABSENT);
+        BOOST_CHECK_EQUAL(flags, NO_ENTRY);
+
+        view->AccessCoin(outp);
+        GetCoinsMapEntry(view->map(), value, flags, outp);
+        BOOST_CHECK_EQUAL(value, coin.out.nValue);
+        BOOST_CHECK_EQUAL(flags, 0);
+    }
+
+    // Can't overwrite an entry without specifying that an overwrite is
+    // expected.
+    Coin tmpCoin2{coin};
+    BOOST_CHECK_THROW(
```

Comment:
```bash
test/coins_tests.cpp:1004:39: error: 'tmpCoin2' used after it was moved [bugprone-use-after-move,-warnings-as-errors]
        view->AddCoin(outp, std::move(tmpCoin2), /*possible_overwrite*/ false),
                                      ^
test/coins_tests.cpp:1004:15: note: move occurred here
        view->AddCoin(outp, std::move(tmpCoin2), /*possible_overwrite*/ false),
              ^
test/coins_tests.cpp:1004:39: note: the use happens in a later loop iteration than the move
        view->AddCoin(outp, std::move(tmpCoin2), /*possible_overwrite*/ false),
```

Reply:
Someone correct me if I'm wrong, but I think this is a spurious check. `tmpCoin2` is initialized fresh for every invocation in this function, and then never used again after move. Clang-tidy is throwing this error because the containing function, `TestFlushBehavior()`, is called in a loop below.

Reply:
It may also be because the `BOOST_CHECK_THROW` macro expands to a loop?

Either way, it should be trivial to work around by just replacing `std::move(tmpCoin2)` with `Coin{coin}`.